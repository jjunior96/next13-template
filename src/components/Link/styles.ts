import tw from 'tailwind-styled-components';

export const LinkContainer = tw.a`
  text-xl
  text-link
  group 
  transition-all 
  duration-300 
  ease-in-out
`;

export const LinkContent = tw.span`
  bg-gradient-to-r 
  from-link
  to-link 
  bg-[length:0%_2px] 
  bg-left-bottom 
  bg-no-repeat 
  transition-all 
  duration-500 
  ease-out 
  group-hover:bg-[length:100%_2px]
`;
