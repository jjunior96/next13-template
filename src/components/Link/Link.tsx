import { AnchorHTMLAttributes } from 'react';

import * as S from './styles';

type LinkProps = {
  children: React.ReactNode;
} & AnchorHTMLAttributes<HTMLAnchorElement>;

export const Link = ({ children, href, ...rest }: LinkProps) => {
  return (
    <S.LinkContainer href={href} {...rest}>
      <S.LinkContent>{children}</S.LinkContent>
    </S.LinkContainer>
  );
};
