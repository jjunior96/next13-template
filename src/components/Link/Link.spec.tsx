import { render, screen } from '@testing-library/react';

import { Link } from './Link';

describe('<Link />', () => {
  it('should render default correctly', () => {
    const { container } = render(<Link href="/">Link</Link>);

    const link = screen.getByRole('link', { name: /link/i });

    expect(container).toMatchSnapshot();
    expect(link).toHaveAttribute('href', '/');
  });
});
