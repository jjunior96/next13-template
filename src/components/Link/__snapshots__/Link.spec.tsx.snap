// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render default correctly 1`] = `
<div>
  <a
    class="text-xl text-link group transition-all duration-300 ease-in-out"
    href="/"
  >
    <span
      class="bg-gradient-to-r from-link to-link bg-[length:0%_2px] bg-left-bottom bg-no-repeat transition-all duration-500 ease-out group-hover:bg-[length:100%_2px]"
    >
      Link
    </span>
  </a>
</div>
`;
